src/interpolants.jl

```diff
@@ -0,0 +1,97 @@
+"""
+    hermite_interpolant(t, integrator::DEIntegrator, idxs, deriv)
+
+Calculate Hermite interpolation of derivative `deriv` at time `t` and indices `idxs`
+for `integrator`.
+"""
+function hermite_interpolant(t, integrator::DEIntegrator, idxs, deriv)
+    [hermite_interpolant(τ, integrator, idxs, deriv) for τ in t]
+end
+
+function hermite_interpolant(t::Number, integrator::DEIntegrator, idxs, deriv)
+    Θ = (t - integrator.tprev)/integrator.dt
+    OrdinaryDiffEq.hermite_interpolant(Θ, integrator.dt, integrator.uprev,
```

Comment:
This requires `k1 = f(t,uprev)` and `k2 = f(t+dt,u)`. Why would this be true for the methods which normally don't use Hermite?

Reply:
Oh I missed that this is necessary. Since it is absolutely necessary (in my opinion) to somehow calculate initial interpolants without adding additional interpolation steps (otherwise it is always possible that we end up with a recursion that will fail) we then have to check this for every such method (currently only Verner methods are handled) and then probably redefine own custom interpolants. In the case of Verner methods we could use `k1 = f(t, uprev)` but we then have to calculate `f(t+dt, u)` since this is different from all initial interpolation steps `ki`.

Reply:
I just rethought about this and it seems to me this whole problem is already solved by the changed order of interpolation updates of ODE integrator. I had added these special interpolants at the beginning of my update to the interpolation at a time when the order was different and there was no other way to get it running.
Just tested it with the Mackey-Glass equation which fails on master and apparently these special interpolants are not required anymore (at least I could not reproduce any errors). So even better, we can completely get rid of these special treatments and it should just work.